wgpu-core/src/validation.rs

```diff
@@ -147,8 +147,11 @@ pub enum BindingError {
         binding: naga::AddressSpace,
         shader: naga::AddressSpace,
     },
-    #[error("Buffer structure size {0}, added to one element of an unbound array, if it's the last field, ended up greater than the given `min_binding_size`")]
-    WrongBufferSize(wgt::BufferSize),
+    #[error("Buffer structure size {shader}, added to one element of an unbound array, if it's the last field, ended up greater than the given `min_binding_size`, which is {binding}")]
+    WrongBufferSize {
+        binding: wgt::BufferSize,
+        shader: wgt::BufferSize,
+    },
```

Comment:
those field names don't make sense to me. Both are sizes and not binding indices as the names imply

```suggestion
    #[error("Buffer structure size {structure_size}, added to one element of an unbound array, if it's the last field, ended up greater than the given `min_binding_size`, which is {min_binding_size}")]
    WrongBufferSize {
        structure_size: wgt::BufferSize,
        min_binding_size: wgt::BufferSize,
    },
```